_config/config.yml

```diff
@@ -13,10 +13,8 @@ MySQLDatabase:
   collation: utf8_general_ci
 HTTP:
   cache_control:
-    max-age: 0
     must-revalidate: "true"
-    no-transform: "true"
-  vary: "Cookie, X-Forwarded-Protocol, User-Agent, Accept"
```

Comment:
Isn't this an implicit API change? There'll be configurations which relied on those defaults, however bad they might be. At least this will need a pretty solid upgrading guide.

- Removing `Vary: Cookie`: Leaving this out could be dangerous unless HTTP proxies implement "safe but not spec compliant behaviour", namely that they ignore caching instructions if detecting cookie that aren't whitelisted, regardless of `Vary` (CloudFlare and Incapsula do that). The proper answer is not to send `Cache-Control` headers in the first place with those cookies, so `Vary` doesn't apply. I think as long as we stick with "PHP session = potentially private content", we're good. There's some edge cases around cookie-based a/b testing, but I think they can be covered in the upgrading guide rather than 100% backwards compat.
- Removing `Vary: Accept`: That seems fine, it's auto-added by mod_gzip etc anyway
- Removing `Vary: X-Forwarded-Protocol`: Do you have a reference on why this is bad practice? It seems common practice for a cache to include the full URI (incl. protocol) in computing the cache key, making this header redundant. It's more relevant for the application processing a request.
- Adding `Vary: X-Requested-With`: That seems overkill as a default? It's fairly uncommon to vary responses between ajax and non-ajax requests. We might do this in the CMS, but those are never cacheable requests anyway.

Reply:
They may be an implicit API change and yes some sites may rely on these, but this combination is practically "no-cache" as far as a proxy is concerned because these Headers are likely to be unique per visitor so if we want to provide meaningful HTTP caches we just can't have this set up.

`Vary: Cookie`: If we denote the content as private using the `private` directive then this vary is not needed. [Fastly explicitly say `Vary: Cookie` is no good](https://www.fastly.com/blog/best-practices-using-vary-header). If we use `Vary: Cookie` we may as well not use `Cache-Control` it seems.

`Vary: Accept`: Note this is different from `Accept-Encoding` (which is added by `mod_deflate` and other server layers that do compression). `Accept` is the mime types the browser understands (eg: `text/html` or `application/json`). I can't seem to find many references online to support using `Vary: Accept` anywhere.

`Vary: X-Forwarded-Protocol`: This is a header that should only be set by proxies, load balancers, CDNs and is not a header the browser sends, this means it provides no value as a `Vary`.

`Vary: X-Requested-With`: I've added that purely because of our `HTTP.cache_ajax_requests` config option as it seems sensible; though I think you're right that it's not needed... Happy to remove it if you think even with our `HTTP.cache_ajax_requests` option it is not needed.

Reply:
> I would only add this Vary header if its turned off - because that's the only time the behaviour "varies" based on that metadata.

That's true, so I'm happy to take it out, it does make managing the `Vary` stuff a bit harder, though